### ISSUE TITLE
Pledge amounts a11y issue/clarity

### ISSUE THREAD

**Issue by bkardell** [opencollective/opencollective-frontend#5196]
In running our open-prioritization experiment we got feedback that people were unclear on how to pledge a custom amount (it is possible, but not clear) - and that pledge amounts aren't accessible.  It seems like this could (maybe should) just use radio buttons - but if not, aria-pressed I think was suggested by @aardrian should be set.  Idk, if someone tells me how they prefer to proceed I'm happy to send a PR here, or open collective sort this out?

**Comment by aardrian** [comment]
At its simplest, use radio buttons (IMO).

If you keep the buttons, the group of buttons needs a group label, the buttons need to indicate their toggleness, and the number input should be selectable with mouse, not just keyboard.

An example structure:

```
<div role="group" aria-labelledby="GrpLbl">
 <label for="amount" id="GrpLbl">Amount (USD)</label>
 <button aria-pressed="false">$5</button>
 <button aria-pressed="false">$10</button>
 <button aria-pressed="false">$15</button>
 <button aria-pressed="false">$20</button>
 <button aria-pressed="false">$50</button>
 <button aria-pressed="true">$100</button>
 <button aria-pressed="false">$250</button>
 <input type="number" id="amount" value="100">
</div>
```

You can hang styles off `button[aria-pressed="true"]` to ensure the visual style matches the programmatic state.

The `group` role will announce to a screen reader that the buttons are related and provide a group label for them (via `aria-labelledby`).

For narrower viewports (see the following screen shot) don't let the adjacent content overlap the text area — I only discovered it was editable _after_ I donated, meaning I donated at _$50 less_ than I wanted to.

![2020-10-02 (3)](https://user-images.githubusercontent.com/1376607/94982319-863d3400-0507-11eb-84d8-c876cb893192.png)

**Comment by sbinlondon** [comment]
Hi @bkardell please open issues on our main repository here: https://github.com/opencollective/opencollective
